test/gtest/interpolate.hpp

```diff
@@ -0,0 +1,378 @@
+    }
+
+    void RunTest()
+    {
+        auto&& handle = get_handle();
+
+        miopenStatus_t status;
+
+        size_t nelems = output.desc.GetElementSize();
+
+        cpu_interpolate_forward<T>(input, ref_output, nelems, scale_factors, align_corners, mode);
+
+        if(mode == MIOPEN_INTERPOLATE_MODE_NEAREST)
+        {
+            status = miopen::InterpolateNearestForward(handle,
+                                                       input.desc,
+                                                       input_dev.get(),
+                                                       output.desc,
+                                                       output_dev.get(),
+                                                       scale_factors.desc,
+                                                       scale_factors_dev.get(),
+                                                       mode);
+        }
+        else
+        {
+            status = miopen::InterpolateLinearCubicForward(handle,
+                                                           input.desc,
+                                                           input_dev.get(),
+                                                           output.desc,
+                                                           output_dev.get(),
+                                                           scale_factors.desc,
+                                                           scale_factors_dev.get(),
+                                                           mode,
+                                                           align_corners);
+        }
+        fflush(stdout);
+        ASSERT_EQ(status, miopenStatusSuccess);
+
+        output.data = handle.Read<T>(output_dev, output.data.size());
+    }
+
+    void Verify()
+    {
+        double threshold = std::numeric_limits<T>::epsilon();
+
+        auto error = miopen::rms_range(ref_output, output);
+
+        ASSERT_EQ(miopen::range_distance(ref_output), miopen::range_distance(output));
+        EXPECT_LT(error, threshold * 10) << "Error output beyond tolerance Error:" << error
+                                         << ",  Thresholdx10: " << threshold * 10;
+    }
+    InterpolateTestCase interpolate_config;
+
+    tensor<T> input;
+    tensor<T> output;
+    tensor<T> ref_output;
+    tensor<float> scale_factors;
+
+    miopenInterpolateMode_t mode;
+    bool align_corners;
+
+    miopen::Allocator::ManageDataPtr input_dev;
+    miopen::Allocator::ManageDataPtr output_dev;
+    miopen::Allocator::ManageDataPtr scale_factors_dev;
+};
+
+// BACKWARD TEST
+template <typename T = float>
+struct InterpolateTestBwd : public ::testing::TestWithParam<InterpolateTestCase>
+{
+protected:
+    void SetUp() override
+    {
+        auto&& handle      = get_handle();
+        interpolate_config = GetParam();
+
+        auto in_dim      = interpolate_config.GetInput();
+        auto in_grad_dim = in_dim;
```

Comment:
2 variables is similar.

Reply:
in_dim means dim was taken from input, then I paste it to other variable to make variable's name make sense